qiskit/providers/ibmq/job/ibmqjob.py

```diff
@@ -49,157 +51,149 @@ class IBMQJob(BaseJob):
 
         except JobError as ex:
             print("Something wrong happened!: {}".format(ex))
 
-    A call to ``status()`` can raise if something happens at the API level that
-    prevents Qiskit from determining the status of the job. An example of this
-    is a temporary connection lose or a network failure.
+    A call to ``status()`` can raise if something happens at the server level
+    that prevents Qiskit from determining the status of the job. An example of
+    this is a temporary connection lose or a network failure.
 
-    The ``submit()`` and ``status()`` methods are examples of non-blocking API.
-    ``Job`` instances also have `id()` and ``result()`` methods which will
-    block::
+    The ``status()`` method is an example of non-blocking API.
+    The ``result()`` method is an example of blocking API:
 
         job = IBMQJob(...)
-        job.submit()
 
         try:
-            job_id = job.id() # It will block until completing submission.
-            print('The job {} was successfully submitted'.format(job_id))
+            job.submit()
+            print('The job {} was successfully submitted'.format(job.job_id()))
 
             job_result = job.result() # It will block until finishing.
             print('The job finished with result {}'.format(job_result))
 
         except JobError as ex:
             print("Something wrong happened!: {}".format(ex))
 
-    Both methods can raise if something ath the API level happens that prevent
-    Qiskit from determining the status of the job.
+    Many of the ``IBMQJob`` methods can raise ``JobError`` if unexpected
+    failures happened at the server level.
 
     Note:
-        When querying the API for getting the status, two kinds of errors are
-        possible. The most severe is the one preventing Qiskit from getting a
-        response from the backend. This can be caused by a network failure or a
-        temporary system break. In these cases, calling ``status()`` will raise.
+        When querying the server for getting the job information, two kinds
+        of errors are possible. The most severe is the one preventing Qiskit
+        from getting a response from the server. This can be caused by a
+        network failure or a temporary system break. In these cases, the job
+         method will raise.
 
         If Qiskit successfully retrieves the status of a job, it could be it
         finished with errors. In that case, ``status()`` will simply return
         ``JobStatus.ERROR`` and you can call ``error_message()`` to get more
         info.
 
-    Attributes:
-        _executor (futures.Executor): executor to handle asynchronous jobs
     """
-    _executor = futures.ThreadPoolExecutor()
 
-    def __init__(self, backend, job_id, api, qobj=None,
-                 creation_date=None, api_status=None,
-                 use_object_storage=False, use_websockets=False):
+    def __init__(self,
+                 backend_obj: Optional[BaseBackend],
```

Comment:
Trying to figure out if having `Optional` is a typo or maybe I'm missing something. In what cases would `backend_obj` be `None`?

Reply:
It's `Optional` because of `CircuitJob`, which is a child of `IBMQJob`. In `CircuitsManager`, `CircuitJob` is initialized without a backend. I don't want to make it optional since it's expected for all but circuit jobs. Let me discuss with @diego-plan9 .

Reply:
To be reevaluated during #363